# MPAL-hg38

This repository contains code used to re-process the healthy bone marrow cell
samples generated by Granja et al. (2019), using the hg38 genome and cellranger-atac
version 2.0.0. The original publication mapped reads to hg19 using cellranger-atac
version 1.0.0.

Paper: https://www.nature.com/articles/s41587-019-0332-7  
GitHub: https://github.com/GreenleafLab/MPAL-Single-Cell-2019  
GEO: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE139369  
SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRP226885  

## Instructions

The data required is stored on AWS S3 (at the SRA), and the data egress is
**not free**. To download the raw data, you will need to set up an AWS account
and configure `awscli` to use your account.

The workflow requires snakemake, so if you don't have snakemake
installed you can create a new conda environment with snakemake using the
following command:

```
mamba env create -f environment.yaml
conda activate mpal
```

Run the snakemake workflow:

```
snakemake -j 1
```

This will download cellranger-atac 2.0.0, the hg38 reference genome, the raw
data from SRA, and map the reads to hg38. The final output will be
a scATAC-seq fragment file for each sample.